x-pack/filebeat/input/azureblobstorage/input_test.go

```diff
@@ -392,6 +392,132 @@ func Test_StorageClient(t *testing.T) {
 	}
 }
 
+func Test_Concurrency(t *testing.T) {
+	tests := []struct {
+		name        string
+		baseConfig  map[string]interface{}
+		mockHandler func() http.Handler
+		expectedLen int
+	}{
+		{
+			name: "TestConcurrency_100_Workers",
+			baseConfig: map[string]interface{}{
+				"account_name":                        "beatsblobnew",
+				"auth.shared_credentials.account_key": "7pfLm1betGiRyyABEM/RFrLYlafLZHbLtGhB52LkWVeBxE7la9mIvk6YYAbQKYE/f0GdhiaOZeV8+AStsAdr/Q==",
+				"max_workers":                         100,
+				"poll":                                true,
+				"poll_interval":                       "10s",
+				"containers": []map[string]interface{}{
+					{
+						"name": mock.ConcurrencyContainer,
+					},
+				},
+			},
+			mockHandler: mock.AzureConcurrencyServer,
+			expectedLen: mock.TotalRandomDataSets,
+		},
+		{
+			name: "TestConcurrency_1000_Workers",
+			baseConfig: map[string]interface{}{
+				"account_name":                        "beatsblobnew",
+				"auth.shared_credentials.account_key": "7pfLm1betGiRyyABEM/RFrLYlafLZHbLtGhB52LkWVeBxE7la9mIvk6YYAbQKYE/f0GdhiaOZeV8+AStsAdr/Q==",
+				"max_workers":                         1000,
+				"poll":                                true,
+				"poll_interval":                       "10s",
+				"containers": []map[string]interface{}{
+					{
+						"name": mock.ConcurrencyContainer,
+					},
+				},
+			},
+			mockHandler: mock.AzureConcurrencyServer,
+			expectedLen: mock.TotalRandomDataSets,
+		},
+		{
+			name: "TestConcurrency_2000_Workers",
+			baseConfig: map[string]interface{}{
+				"account_name":                        "beatsblobnew",
+				"auth.shared_credentials.account_key": "7pfLm1betGiRyyABEM/RFrLYlafLZHbLtGhB52LkWVeBxE7la9mIvk6YYAbQKYE/f0GdhiaOZeV8+AStsAdr/Q==",
+				"max_workers":                         2000,
+				"poll":                                true,
+				"poll_interval":                       "500s",
```

Comment:
Is this interval different in this case for a reason?

Reply:
you can ignore this, will revert it to 10s like the others